pkg/subscribe/watcher_test.go

```diff
@@ -0,0 +1,164 @@
+package subscribe
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/schemas"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_stream(t *testing.T) {
+	tests := []struct {
```

Comment:
Nit: Might also be good to include test cases for the error scenarios (no schema/store doesn't support watches/access control denied request). However, you didn't change that code here, so I'm ok with not including that.

Reply:
Added some error test cases